libs/game/effects.ts

```diff
@@ -57,6 +57,9 @@ namespace effects {
         //% group="Images"
         startSceneEffect(times?: number, delay?: number): void {
             times = times ? times : 15;
+
+            // fill in colors so that the effects apply to entire background
+            scene.backgroundImage().replace(0, scene.backgroundColor());
```

Comment:
This always creates an backagroubd image. What happens when there isn’t an image?

Reply:
``scene.backgroundImage()`` calls ``scene.Background.image`` which will create an image if one doesn't exist; without this https://github.com/Microsoft/pxt-common-packages/blob/387a22c48292cb5750d980f019bff7dd97d63c09/libs/game/effects.ts#L65

would still create one, but not include the background color when modifying the image. I can add a way to check if is currently a background image in `scene.Background` and just not do anything if there isn't one, but that seems confusing.

related, https://github.com/Microsoft/pxt-common-packages/blob/387a22c48292cb5750d980f019bff7dd97d63c09/libs/game/background.ts#L48 should probably be changed to `scene.drawTransparentImage` to let background color appear through clear spots in the background image, correct?

Reply:
yes it should use drawTransparentImage.